### PR TITLE
Flatten a bunch of non-pervasive ParseNode types (part 1)

### PR DESCRIPTION
**Pull request by marcianx**
Towards #1492.
Also, this renames some of the `value` sub-fields to something more descriptive.

### CODE REVIEW

File: src/parseNode.js [hunk 1]
```diff
@@ -418,30 +370,19 @@ type ParseNodeTypes = {
...
     "vphantom": {|
         type: "vphantom",
         mode: Mode,
         loc?: ?SourceLocation,
-        value: {|
-            type: "vphantom",
-            body: AnyParseNode,
-            value: AnyParseNode[],
-        |},
+        body: AnyParseNode,
```
Comment (kevinbarabash) [inline src/parseNode.js]: Why don't we need the `AnyParseNode[]` that was originally on the phantoms?

File: src/parseNode.js [hunk 2]
```diff
@@ -344,57 +316,37 @@ type ParseNodeTypes = {
...
         type: "mclass",
         mode: Mode,
         loc?: ?SourceLocation,
-        value: {|
-            type: "mclass",
-            mclass: string,
-            value: AnyParseNode[],
-        |},
-    |},
-    "mod": {|
```
Comment (kevinbarabash) [inline src/parseNode.js]: It's nice that we can get rid whole nodes, but where did `mod` go?

File: src/functions/mathchoice.js [hunk 3]
```diff
@@ -6,19 +6,16 @@ import Style from "../Style";
...
+import type {ParseNode} from "../parseNode";
 
+const chooseMathStyle = (group: ParseNode<"mathchoice">, options) => {
+    switch (options.style.size) {
+        case Style.DISPLAY.size: return group.display;
+        case Style.TEXT.size: return group.text;
+        case Style.SCRIPT.size: return group.script;
+        case Style.SCRIPTSCRIPT.size: return group.scriptscript;
+        default: return group.text;
+    }
```
Comment (kevinbarabash) [inline src/functions/mathchoice.js]: nice cleanup

File: src/functions/delimsizing.js [hunk 4]
```diff
@@ -134,11 +127,10 @@ defineFunction({
 });
 
 
-function leftRightGroupValue(group: ParseNode<"leftright">): LeftRightDelimType {
-    if (!group.value.body) {
+function assertParsed(group: ParseNode<"leftright">) {
```
Comment (kevinbarabash) [inline src/functions/delimsizing.js]: I guess since we're in delimsizing.js this can be a more general name.

File: src/parseNode.js (hunk 2 above)
Comment (edemaine) [reply src/parseNode.js]: `mod` was replaced with macros a while back. I guess I forgot to remove the type.

File: src/parseNode.js (hunk 1 above)
Comment (marcianx) [reply src/parseNode.js]: `value = ordargument(body)`. There was no reason to store both.

Plus I've gotten allergic to anything named `value`. :) We should probably rename the rest to have names that say what they are.

**Pull request merged**